posts/serializers.py

```diff
@@ -181,3 +195,15 @@ def create(self, validated_data):
 
```

Comment:
The following is a way to actually encode the url and source on creation for local posts so we don't have to use your functions. Up to you whether youd rather just keep what you have...

```
if post.source =="":
    post.source=SITE_URL + 'posts/' + str(post.id)
    post.origin=SITE_URL + 'posts/' + str(post.id)
```

Reply:
I like keeping them blank on local posts. Kind of weird but it allows us to assume if the post.source is blank we know it has to be local